docs/src/passing-in-data.rst

```diff
@@ -83,15 +84,66 @@ which gets the value of the environment variable **MYBUCKET**.
 Getting a Value from an SSM Store Variable
 ==========================================
 
-To get the value of an SSM parameter store variable,
-use code like the following,
-which uses the value of the SSM variable **my-awesome-value**.
+There are three ways to get the value of an SSM parameter store variable, depending on whether you want
+the latest version of a plain string, a particular version of a plain string, or a particular version
+of a secret string. It is not possible to retrieve the latest version of a secure string. To read the
+latest version of a secret, you have to read the secret from SecretsManager (see `doc:using_value_from_secrets_manager`).
+
+To retrieve the latest value once (as a context value, see the :doc:`context` topic), and keep on using the same value
+until the context value manually refreshed, use a :py:class:`SSMParameterProvider <@aws-cdk/cdk.SSMParameterProvider>`:
+
+.. code-block:: ts
+
+    const myvalue = new SSMParameterProvider(this).getString("my-parameter-name");
+
+To read a particular version of an SSM Parameter Store plain string value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreString <@aws-cdk/cdk.SsmParameterStoreString>`:
+
+.. code-block:: ts
+
+    const parameterString = new SsmParameterStoreString(this, 'MyParameter', {
+        parameterName: 'my-parameter-name',
+        version: 1,
+    });
+    const myvalue = parameterString.value;
+
+To read a particular version of an SSM Parameter Store SecureString value at CloudFormation deployment time,
+use :py:class:`SsmParameterStoreSecureString <@aws-cdk/cdk.SsmParameterStoreSecureString>`:
+
+.. code-block:: ts
+
+    const secureString = new SsmParameterStoreSecureString(this, 'MySecretParameter', {
+        parameterName: 'my-secret-parameter-name',
+        version: 1,
+    });
+    const myvalue = secureString.value;
+
+
+.. _using_value_from_secrets_manager:
+
+Getting a Value from AWS Secrets Manager
+========================================
+
+To use values from AWS Secrets Manager in your CDK app, create an instance of :py:class:`SecretsManagerValue
+<@aws-cdk/cdk.SecretsManagerValue>`. It represents a value that is retrieved from Secrets Manager and used
+at CloudFormation deployment time.
 
 .. code-block:: ts
 
-    const myvalue = new SSMParameterProvider(this).getString("my-awesome-value");
```

Comment:
We should still document SSMParameterProvider I suppose, no?

Reply:
It is, the diff is just whack.

Reply:
okay!